src/CircuitSimulator.cpp

```diff
@@ -182,11 +183,11 @@ std::map<std::size_t, bool> CircuitSimulator<DDPackage>::single_shot(const bool
                     }
                 }
             }
-            Simulator<DDPackage>::dd->garbageCollect();
+            Simulator<Config>::dd->garbageCollect();
```

Comment:
at some points in the code you refactored to using `this`, in some you did not. Wouldn't hurt to be consistent throughout.